exchanges/lbank/lbank_wrapper.go

```diff
@@ -372,25 +372,32 @@ func (l *Lbank) ModifyOrder(action *order.Modify) (string, error) {
 }
 
 // CancelOrder cancels an order by its corresponding ID number
-func (l *Lbank) CancelOrder(order *order.Cancel) error {
-	fpair, err := l.FormatExchangeCurrency(order.Pair, order.AssetType)
+func (l *Lbank) CancelOrder(o *order.Cancel) error {
+	if err := o.Validate(o.StandardCancel()); err != nil {
+		return err
+	}
+	fpair, err := l.FormatExchangeCurrency(o.Pair, o.AssetType)
 	if err != nil {
 		return err
 	}
-	_, err = l.RemoveOrder(fpair.String(), order.ID)
+	_, err = l.RemoveOrder(fpair.String(), o.ID)
 	return err
 }
 
 // CancelAllOrders cancels all orders associated with a currency pair
-func (l *Lbank) CancelAllOrders(orders *order.Cancel) (order.CancelAllResponse, error) {
+func (l *Lbank) CancelAllOrders(o *order.Cancel) (order.CancelAllResponse, error) {
```

Comment:
from what i can understand all these validates are checking if the required/necessary params are filled properly. In case of this function, the way its set out asset type is a must (otherwise FormatExchangeCurrency will output an error) so either the function needs to change or asset type needs to be passed in. idk what the best outcome is, but if an empty order.Cancel{} is given to the function it errors on format exch currecny

Reply:
Doesn't have format exchange currency in this function from what I can tell but will add a currency pair and asset type check regardless as a baseline validation check.